### PR TITLE
Add GITOID_URI_SPEC

### PR DESCRIPTION
**Pull request by edwarnicke**
Signed-off-by: Ed Warnicke <hagbard@gmail.com>

### CODE REVIEW

**Comment by edwarnicke** [comment]
See also: https://github.com/git-bom/spec/issues/2

File: GITOID_URI_SPEC.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+Current valid values for <git object type> are 'blob','tree','commit', and 'tag'.
+Current valid values for <hash algorithm> are 'sha1' and 'sha256'
+<hash value> should be expressed as a hexidecimal string in lower case.
+
+Example: gitoid:blob:sha1:261eeb9e9f8b2b4b0d119366dda99c6fd7d35c64
+
+Scheme semantics:
+
+gitoid stands for Git Object ID.  Git is an object store.  It currently has four types of objects.  A git object is formed by prepending an object header to the contents of the object.  A git object header consists of:
```
Comment (AevaOnline) [inline GITOID_URI_SPEC.md:29]:
```suggestion
gitoid stands for Git Object ID.  Git is an object store, which currently supports four types of objects.  A git object is formed by prepending an object header to the contents of the object.  A git object header consists of:
```

File: GITOID_URI_SPEC.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+https://git-scm.com/book/en/v2/Git-Internals-Git-Objects
+
+Scheme syntax: gitoid":"<git object type>":"<hash algorithm>":"<hash value>
+
+Current valid values for <git object type> are 'blob','tree','commit', and 'tag'.
+Current valid values for <hash algorithm> are 'sha1' and 'sha256'
+<hash value> should be expressed as a hexidecimal string in lower case.
+
+Example: gitoid:blob:sha1:261eeb9e9f8b2b4b0d119366dda99c6fd7d35c64
+
```
Comment (AevaOnline) [inline GITOID_URI_SPEC.md:26]: would it be worth noting here that, at present, our schema only recognizes using one of the four types ('blob') or that we may add additional types (e.g., 'bom')? or do you think this information should be captured elsewhere?

File: GITOID_URI_SPEC.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+Applications/protocols that use this scheme name:
+     See [Section 3.5](https://www.rfc-editor.org/rfc/rfc7595.html#section-3.5).
+Contact:
+     Registering party: Ed Warnicke <eaw@cisco.com>
+
+Scheme Creator: GitBOM
+
+Change controller:
+    Either the registering party or someone who is verified to represent
```
Comment (fkautz) [inline GITOID_URI_SPEC.md:13]: Should reword to: "Change controller: Either the registering party or someone verified to represent the scheme creator. See the previous answer."

File: GITOID_URI_SPEC.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    the scheme creator.  See previous answer.
+
+References:
+https://git-scm.com/book/en/v2/Git-Internals-Git-Objects
+
+Scheme syntax: gitoid":"<git object type>":"<hash algorithm>":"<hash value>
+
+Current valid values for <git object type> are 'blob','tree','commit', and 'tag'.
+Current valid values for <hash algorithm> are 'sha1' and 'sha256'
+<hash value> should be expressed as a hexidecimal string in lower case.
```
Comment (fkautz) [inline GITOID_URI_SPEC.md:23]: s/hexidecimal/hexadecimal/

File: GITOID_URI_SPEC.md (hunk 1 above)
Comment (fkautz) [inline GITOID_URI_SPEC.md:29]: s/contents of the object/object's contents/

File: GITOID_URI_SPEC.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+Example: gitoid:blob:sha1:261eeb9e9f8b2b4b0d119366dda99c6fd7d35c64
+
+Scheme semantics:
+
+gitoid stands for Git Object ID.  Git is an object store.  It currently has four types of objects.  A git object is formed by prepending an object header to the contents of the object.  A git object header consists of:
+
+<git object type>" "<size of contents as decimal string>"\0"
+
+The git object id is computed by applying a hash to the git object.  Currently the most common hash is sha1, but there is some effort to transition to sha256 over time.
```
Comment (fkautz) [inline GITOID_URI_SPEC.md:33]: s/Currently/Currently,/

Comment (fkautz) [reply GITOID_URI_SPEC.md:33]: Also, this line may end up changing... e.g. if we go with the unified line where we have more than one hash. Might be `sha1+sha256` or some similar derivative

**Review by fkautz — CHANGES_REQUESTED** [review]

File: GITOID_URI_SPEC.md (hunk 2 above)
Comment (edwarnicke) [reply GITOID_URI_SPEC.md:26]: I think we put this in as a general schema for gitoids, which we then use.  Not as a specific gitbom schema.

File: GITOID_URI_SPEC.md (hunk 1 above)
Comment (edwarnicke) [reply GITOID_URI_SPEC.md]: I believe I've fixed this.

File: GITOID_URI_SPEC.md (hunk 3 above)
Comment (edwarnicke) [reply GITOID_URI_SPEC.md]: I believe I've fixed this.

File: GITOID_URI_SPEC.md (hunk 4 above)
Comment (edwarnicke) [reply GITOID_URI_SPEC.md]: I believe I've fixed this.

File: GITOID_URI_SPEC.md (hunk 5 above)
Comment (edwarnicke) [reply GITOID_URI_SPEC.md]: > s/Currently/Currently,/

I believe I've fixed this.

Comment (edwarnicke) [reply GITOID_URI_SPEC.md]: > Also, this line may end up changing... e.g. if we go with the unified line where we have more than one hash. Might be `sha1+sha256` or some similar derivative

Let's leave this be for the moment.

File: GITOID_URI_SPEC.md (hunk 1 above)
Comment (edwarnicke) [reply GITOID_URI_SPEC.md]: I believe I've fixed this.

**Review by nellshamrell — APPROVED** [review]
:shipit:

**Pull request merged**